go-apps/meep-app-enablement/server/app-support/app-support.go

```diff
@@ -316,3 +369,206 @@ func populateServiceInfoList(key string, jsonInfo string, sInfoList interface{})
 	data.ServiceInfos = append(data.ServiceInfos, sInfo)
 	return nil
 }
+
+func applicationsSubscriptionsPOST(w http.ResponseWriter, r *http.Request) {
+
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+
+	vars := mux.Vars(r)
+	appInstanceId := vars["appInstanceId"]
+
+	//check if entry exist for the application in the DB
+	key := appEnablementBaseKey + ":apps:" + appInstanceId
+	fields, err := rc.GetEntry(key)
+	if err != nil || len(fields) == 0 {
+		log.Error("AppInstanceId does not exist, app is not running")
+		http.Error(w, "AppInstanceId does not exist, app is not running", http.StatusBadRequest)
+		return
+	}
+
+	var subscription AppTerminationNotificationSubscription
+	decoder := json.NewDecoder(r.Body)
+	err = decoder.Decode(&subscription)
+	if err != nil {
+		log.Error(err.Error())
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	//checking for mandatory properties
+	if subscription.CallbackReference == "" {
+		log.Error("Mandatory CallbackReference parameter not present")
+		http.Error(w, "Mandatory CallbackReference parameter not present", http.StatusBadRequest)
+		return
+	}
+
+	if subscription.SubscriptionType != APP_TERMINATION_NOTIFICATION_SUBSCRIPTION_TYPE {
+		log.Error("SubscriptionType shall be AppTerminationNotificationSubscription")
+		http.Error(w, "SubscriptionType shall be AppTerminationNotificationSubscription", http.StatusBadRequest)
+		return
+	}
+
+	if subscription.AppInstanceId == "" {
+		log.Error("Mandatory AppInstanceId parameter not present")
+		http.Error(w, "Mandatory AppInstanceId parameter not present", http.StatusBadRequest)
+		return
+	}
+
+	if subscription.AppInstanceId != appInstanceId {
+		log.Error("AppInstanceId in endpoint and in body not matching")
+		http.Error(w, "AppInstanceId in endpoint and in body not matching", http.StatusBadRequest)
+		return
+	}
+
+	newSubsId := nextSubscriptionIdAvailable
+	nextSubscriptionIdAvailable++
+	subsIdStr := strconv.Itoa(newSubsId)
+
+	link := new(Self)
+	self := new(LinkType)
+	self.Href = hostUrl.String() + basePath + "applications/" + appInstanceId + "/subscriptions/" + subsIdStr
+	link.Self = self
+	subscription.Links = link
+
+	//registration
+	registerAppTerm(&subscription, newSubsId)
```

Comment:
Why register the subscription here? Don't we have the necessary info in the subscription we are storing in redis?

Reply:
same as previous comment... it populates the map to minimize queries to the DB (although same info is present in both)